src/main/java/at/hannibal2/skyhanni/features/dungeon/DungeonFinderFeatures.kt

```diff
@@ -1,189 +1,360 @@
+                    if (pricePattern.matches(note) && carryPattern.matches(note)) {
+                        highlightParty[slot] = LorenzColor.RED
+                        continue
+                    }
+                }
 
-            if (config.markIneligibleGroups && stack.getLore().any { ineligiblePattern.matches(it) }) {
-                slot highlight LorenzColor.DARK_RED
-                continue
-            }
+                if (config.markNonPugs) {
+                    val note = lore.filter { notePattern.matches(it) }.joinToString(" ").uppercase()
 
-            if (config.markPaidCarries) {
-                val note = stack.getLore().filter { notePattern.containsMatchIn(it) }.joinToString(" ")
+                    if (nonPugPattern.matches(note)) {
+                        highlightParty[slot] = LorenzColor.LIGHT_PURPLE
+                        continue
+                    }
+                }
 
-                if (pricePattern.containsMatchIn(note) && carryPattern.containsMatchIn(note)) {
-                    slot highlight LorenzColor.RED
+                val members = lore.filter {
+                    memberPattern.matches(it)
+                }
+                val memberLevels = members.map {
+                    memberPattern.matchMatcher(it) {
+                        group("level").toInt()
+                    }
+                }
+                val memberClasses = members.map {
+                    memberPattern.matchMatcher(it) {
+                        group("className")
+                    }
+                }
+                if (memberLevels.any { (it ?: Integer.MAX_VALUE) <= config.markBelowClassLevel }) {
+                    highlightParty[slot] = LorenzColor.YELLOW
                     continue
                 }
-            }
-
-            if (config.markNonPugs) {
-                val note = slot.stack.getLore().filter { notePattern.containsMatchIn(it) }.joinToString(" ")
 
-                if (nonPugPattern.containsMatchIn(note)) {
-                    slot highlight LorenzColor.LIGHT_PURPLE
-                    continue
+                if (config.markMissingClass && memberClasses.none { it == selectedClass }) {
+                    highlightParty[slot] = LorenzColor.GREEN
                 }
             }
+        }
+    }
 
-            val members = slot.stack.getLore().filter { memberPattern.matches(it) }
-            val memberLevels = members.map { memberPattern.matchEntire(it)?.groupValues?.get(2)?.toInt() ?: 0 }
-            val memberClasses = members.map { memberPattern.matchEntire(it)?.groupValues?.get(1) ?: "" }
-
-            if (memberLevels.any { it <= config.markBelowClassLevel }) {
-                slot highlight LorenzColor.YELLOW
-                continue
+    private fun toolTipHandler(event: InventoryFullyOpenedEvent) {
+        val inventoryName = event.inventoryName
+        if (partyFinderTitlePattern.matches(inventoryName)) {
```

Comment:
Also remove the closing bracket
```suggestion
        if (!partyFinderTitlePattern.matches(inventoryName)) return
```